core/chain_manager.go

```diff
@@ -525,10 +529,27 @@ func (self *ChainManager) InsertChain(chain types.Blocks) (int, error) {
 		stats      struct{ queued, processed, ignored int }
 		tstart     = time.Now()
 	)
+
+	// check the nonce in parallel to the block processing
+	// this speeds catching up significantly
+	errch := make(chan error, len(chain))
+	go func() {
+		for _, block := range chain {
+			go func(block *types.Block) {
+				if !self.pow.Verify(block) {
+					errch <- ValidationError("Block's nonce is invalid (= %x)", block.Nonce)
+				} else {
+					errch <- nil
+				}
+			}(block)
+		}
+	}()
+
```

Comment:
I'll do some benchmarking and check if it makes any difference. Go is pretty good at scheduling. If none the much more simplistic version will stay.